hls4ml/templates/vivado_template.py

```diff
@@ -71,6 +71,33 @@
     using product = nnet::product::{product_type}<x_T, y_T, res_T>;
 }};\n"""
 
+rnn_mult1_config_template = """struct config{index}_mult1 : nnet::dense_config {{
+    static const unsigned n_in = {n_in};
+    static const unsigned n_out = {n_out};
+    static const unsigned io_type = nnet::{iotype};
+    static const unsigned strategy = nnet::{strategy};
+    static const unsigned reuse_factor = {reuse};
+    typedef {accum_t} accum_t;
+    typedef {bias_t} bias_t;
+    typedef {weight_t} weight_t;
+    template<class x_T, class y_T, class res_T>
+    using product = nnet::product::{product_type}<x_T, y_T, res_T>;
+}};\n"""
+
+
+rnn_mult2_config_template = """struct config{index}_mult2 : nnet::dense_config {{
+    static const unsigned n_in = {n_in};
+    static const unsigned n_out = {n_out};
+    static const unsigned io_type = nnet::{iotype};
+    static const unsigned strategy = nnet::{strategy};
+    static const unsigned reuse_factor = {reuse};
+    typedef {accum_t} accum_t;
+    typedef {bias_t} bias_t;
+    typedef {weight_t} weight_t;
+    template<class x_T, class y_T, class res_T>
+    using product = nnet::product::{product_type}<x_T, y_T, res_T>;
+}};\n"""
+
```

Comment:
Since the contents of these are the same as the Dense config, do we need to define new templates? Can you just pick up the existing Dense template?